pkg/nfd-worker/nfd-worker.go

```diff
@@ -214,11 +256,33 @@ func (w *nfdWorker) Run() error {
 			}
 
 			if w.args.SleepInterval > 0 {
-				trigger = time.After(w.args.SleepInterval)
+				labelTrigger = time.After(w.args.SleepInterval)
+			}
+
+		case e := <-configWatch.Events:
+			name := filepath.Clean(e.Name)
+
+			// If any of our paths (directories or the file itself) change
+			if _, ok := paths[name]; ok {
+				// Blindly remove existing watch and add a new one
+				if err := configWatch.Close(); err != nil {
+					stderrLogger.Printf("WARNING: failed to close fsnotify watcher: %v", err)
+				}
+				configWatch, paths, err = addConfigWatch(configFilePath)
+				if err != nil {
+					return err
+				}
+
+				configTrigger = time.After(time.Second)
```

Comment:
I'm not sure to understand why we are not directly reloading the configuration here. Do we need another channel for this?

Reply:
Woops, me neither, anymore 🙄 Very good comment, thanks for pointing this out. Doesn't seem to make any sense, I'll change this

I really don't remember anymore as the first versions date back to last November or so. But, I think this is just some stale leftover ripples from some early prototype that I became blind with.

Reply:
OK, now that I took a more complete look at the code I start to remember 😄  This is actually the grace period / rate limiter of numerous events in quick succession. We actually want this. I'll add a comment

Reply:
Changed to `0`

Reply:
Added a comment explaining this use of double event channels